plugins/inputs/kafka_consumer/kafka_consumer.go

```diff
@@ -221,6 +374,12 @@ func (k *KafkaConsumer) Gather(_ telegraf.Accumulator) error {
 }
 
 func (k *KafkaConsumer) Stop() {
+	if k.topicClient != nil {
+		k.topicClient.Close()
+	}
+	if k.ticker != nil {
+		k.ticker.Stop()
+	}
```

Comment:
Maybe switch position of the two and add some synchronization to prevent a refresh being triggered (or being on-going) while we close the client.

Reply:
Done.